common/test/acceptance/tests/studio/test_import_export.py

```diff
@@ -0,0 +1,144 @@
+class ImportTestMixin(object):
+    """
+    Tests to run for both course and library import pages.
+    """
+    def setUp(self):
+        super(ImportTestMixin, self).setUp()
+        self.import_page = self.import_page_class(*self.page_args())
+        self.landing_page = self.landing_page_class(*self.page_args())
+        self.import_page.visit()
+
+    @abstractmethod
+    def page_args(self):
+        """
+        Generates the args for initializing a page object..
+        """
+        return []
+
+    def test_upload(self):
+        """
+        Ensure uploading of courses works.
+        """
+        self.import_page.upload_tarball(self.tarball_name)
+        self.import_page.wait_for_upload()
+
+    def test_landing_url(self):
+        """
+        Verify that the landing URL goes to the right place.
+        """
+        self.import_page.upload_tarball(self.tarball_name)
+        self.assertEqual(self.import_page.target_url(), self.landing_page.url)
+
+    def test_bad_filename_error(self):
+        """
+        Make sure that the page displays an error if the user attempts to load the wrong file type.
+        """
+        self.import_page.upload_tarball('funny_cat_video.mp4')
+        self.import_page.wait_for_error()
+
+    def test_task_list(self):
+        """
+        Verify that each task item green lights.
+        """
+        # The task list shouldn't be visible to start.
+        self.assertFalse(self.import_page.is_task_list_showing(), "Task list shown too early.")
+        self.import_page.wait_for_tasks()
+        self.import_page.upload_tarball(self.tarball_name)
+        self.import_page.wait_for_tasks(completed=True)
+        self.assertTrue(self.import_page.is_task_list_showing(), "Task list did not display.")
+
+
+class TestCourseImport(ImportTestMixin, StudioCourseTest):
+    """
+    Tests the Course import page
+    """
+    tarball_name = '2015.lzdwNM.tar.gz'
+    import_page_class = ImportCoursePage
+    landing_page_class = CourseOutlinePage
+
+    def page_args(self):
+        return [self.browser, self.course_info['org'], self.course_info['number'], self.course_info['run']]
+
+    def test_course_updated(self):
+        self.landing_page.visit()
+        # Should not exist yet.
+        self.assertRaises(IndexError, self.landing_page.section, "Section")
+        self.import_page.visit()
+        self.import_page.upload_tarball(self.tarball_name)
+        self.import_page.wait_for_upload()
+        self.landing_page.visit()
+        # There's a section named 'Section' in the tarball.
+        self.landing_page.section("Section")
+
+
+class TestLibraryImport(ImportTestMixin, StudioLibraryTest):
+    """
+    Tests the Library import page
+    """
+    tarball_name = 'library.HhJfPD.tar.gz'
+    import_page_class = ImportLibraryPage
+    landing_page_class = LibraryEditPage
+
+    def page_args(self):
+        return [self.browser, self.library_key]
+
+    def test_library_updated(self):
```

Comment:
A test around what happens with a bad component would be good (for a course, you should be able to go to the unit with the error-- for library, just a general error).